### ISSUE TITLE
Fee Payment Confirmation Email Template Updates

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#3142]
#### Description
During testing it was observed that fee payment confirmation email templates were missing key information around the new LDD as well as some tweaks needed to be made to the style of the templates.

#### Dependencies
Are there any dependencies?

#### DOD
- [x] Update Fee Payment Confirmation Email Template when only a deposit is paid on a fee estimate
https://bcgov.sharepoint.com/:w:/t/03521/EXRPViULeKhPswf_6ogI0TQBDmGLqIkeGaxd_8g6xaeYvg?e=xCL6A0

- [x] Update Fee Payment Confirmation Email Template when the full fee estimate is paid
https://bcgov.sharepoint.com/:w:/t/03521/EUGuTjU29NVCrcfyZxwq8O4BhuZ1r36Sth5guFcfEt4rOw?e=Y9Q93a 

- [x] Update Fee Payment Confirmation Email Template for Outstanding Balance
https://bcgov.sharepoint.com/:w:/t/03521/EXBAG5sscxVBjTPofei9FuQBtJd63N43EC81t6G_XaYPJw?e=vK7FYc

Legislated Due Date Recalculation - Acceptance Criteria:

Assumptions: 
- When a request is put on hold from the fee estimate state, successful payment will shift the state back to Call for Records
- When a request is put on hold from the Response state, successful payment will shift the state back to Response
- On hold days will now include the day payment is received
- Payment received on a normal business day, regardless of time, is considered received that day
- Payment received on a weekend or holiday will be considered received the next business day

**Scenario 1: Request put on Hold**
* GIVEN a request is in the Fee Estimate or Response State
* AND there is an approved CFR form with a balance remaining
* WHEN the request state is changed to on hold
* THEN the date the request is put on hold does NOT count against the Legislated Due Date or CFR Due Date
* AND the LDD and CFR countdowns are suspended

**Scenario 2: Payment Received - Business Days**
* GIVEN a general request is in the On-Hold state
* WHEN the request is taken off hold
* AND the date the payment is received falls within a normal business day, regardless of the time
* THEN the legislative due date will be recalculated, by adding the on hold days to the legislative due date prior to the request going on hold
* AND the day the request comes off hold will count as an on-hold day

**Scenario 3: Payment Received - Weekends/Holidays**
* GIVEN a general request is in the On-Hold state
* WHEN the request is taken off hold
* AND the date the payment is received falls on a weekend, or a holiday
* THEN the the legislative due date will be recalculated, by adding the on hold days to the legislative due date prior to the request going on hold
* AND the day the request comes off hold will be considered as the NEXT business day and will count as an on-hold day

**Comment by m-prodan** [comment]
NOTE - Legislated Due Date noted in email templates must be inserted into template after the due date recalculation is done when the request is taken off of hold.

Logic rules for recalculating due date - the day a request goes on hold and the day the request comes off hold and all BUSINESS days between do NOT count against the request countdown, and due date should be adjusted accordingly. 

cc: @abin-aot @lmullane @liseandtea @KyEggleston @JHarrietha-AOT

**Comment by m-prodan** [comment]
I have moved to top of Sprint Backlog, as this has been identified as a critical issue needing to be in place before we can go to PROD

**Comment by divyav-aot** [comment]
@m-prodan I am ready to pick up this task. Please confirm if we can proceed with the changes.

**Comment by lmullane** [comment]
Still waiting for the green light.

**Comment by m-prodan** [comment]
@sumathi-thirumani-aot @divyav-aot - please note, I have had to update the on hold calculation to be consistent with how IAO understands the on hold calculation - namely that the day payment is received is now back to being considered an on-hold day. This will add a day to the examples I went through with Sumathi.

cc @abin-aot

**Comment by JHarrietha-AOT** [comment]
@lmullane I have created bug #3197 due to errors happening in the LDD calculation. When they are not being taken on and off hold multiple times in the same day they are calculation correctly.

We are unable to test scenario 3 as it much have a payment processed over the weekend to test. Let me know if you will have time to do that this weekend. I will leave the ticket in QA in Progress until Scenario 3 can be tested

**Comment by lmullane** [comment]
Thanks for the update, @JHarrietha-AOT.

**Comment by lmullane** [comment]
Hi @JHarrietha-AOT, @abin-aot, @liseandtea 

I tested Scenario 3 and it failed.  I put the request On Hold on December 17th (a Saturday) and paid on the same day. Prior to the request going On Hold the LDD was January 18th.

The new LDD should have been January 19th (i.e. add a day for the next business day that the request came off hold).  But the LDD remained January 18th.

Here is the LDD prior to going On Hold to pay the outstanding balance:
![Screen Shot 2022-12-17 at 7.19.23 PM.png](https://images.zenhubusercontent.com/571e6eb3bf93871ccd9a03d4/fa53660f-b24b-466c-8dba-ae90b1ff5243)

Here is the LDD after payment of the Outstanding Balance:
![Screen Shot 2022-12-17 at 7.39.18 PM.png](https://images.zenhubusercontent.com/571e6eb3bf93871ccd9a03d4/c92a6b11-0d78-4287-b6d1-edcc286ad3c2)

And here is the email to the applicant also stating the wrong LDD:![Screen Shot 2022-12-17 at 7.40.32 PM.png](https://images.zenhubusercontent.com/571e6eb3bf93871ccd9a03d4/116ab747-1b4a-4e4b-a857-88f0c8c920e6)

**Comment by JHarrietha-AOT** [comment]
@lmullane Had a chat with @liseandtea 

In what you outlined above we would not add an additional day. A request that is put on hold on a Saturday (Outside of Business Hours) and Paid on a Saturday (Outside of Business Hours) Would not effect the LDD as all the actions were taken outside of the expected working day for an analyst (they would need to be doing overtime to put a request on hold on weekend which shouldn't could)

If the request was put on hold on a Friday, and then paid on the Saturday an additional day would be added.  

Have updated both @sumathi-thirumani-aot & @divyav-aot to this which they have said is how it is calculating now

**Comment by nkan-aot** [comment]
remaining issue with this ticket has been fixed with #3197, closing this ticket now
